### PR TITLE
define MSRV and check for it in CI

### PR DESCRIPTION
**Pull request by Luap99**
Define 1.76 as our MSRV currently and add a new container image that we use in CI to test that.

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon @baude PTAL

File: README.md [hunk 1]
```diff
@@ -22,6 +22,12 @@ Netavark is a tool for configuring networking for Linux containers. Its features
 - [Podman](https://podman.io/docs) 4.0+
 - [protoc](https://grpc.io/docs/protoc-installation/)
 
+## MSRV (Minimum Supported Rust Version)
+
```
Comment (mheon) [inline README.md:26]: Add a sentence here: "Netavark is guaranteed to build on this Rust version and later." or similar

File: contrib/container_images/build_and_publish_rust_image.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -e
+
+PODMAN=${PODMAN:-podman}
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+DOCKERFILE="$(dirname "${BASH_SOURCE[0]}")/Dockerfile.Rust"
+
+# get the tag from the Dockerfile so we do not duplicate it
+TAG=$(awk -F':' '/FROM/{print $NF}' $DOCKERFILE)
```
Comment (edsantiago) [inline contrib/container_images/build_and_publish_rust_image.sh:10]: What would you think of `if [[ -z "$TAG" ]]; then echo "FOO! empty tag"; exit 1; fi` or equivalent?

File: README.md (hunk 1 above)
Comment (Luap99) [reply README.md:26]: Given rust weird stance on backwards compatibility I find guaranteed to be a very strong word, i.e. it is not so recent that I think rust 1.80 failed to compile older version of the time crate
https://github.com/rust-lang/rust/issues/127343#issuecomment-2218261296

i.e. it is impossible to build aardvark-dns 1.10.0 with rust v1.80 or newer. To their credit they provide a helpful error but still
```
error[E0282]: type annotations needed for `Box<_>`
  --> /home/pholzing/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.30/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
   = note: this is an inference error on crate `time` caused by an API change in Rust 1.80.0; update `time` to version `>=0.3.35` by calling `cargo update`
```

And given we to not test all rust versions there might be some weird stuff in between so I think I rather `s/guaranteed/supposed` in your sentence

File: contrib/container_images/build_and_publish_rust_image.sh (hunk 2 above)
Comment (Luap99) [reply contrib/container_images/build_and_publish_rust_image.sh:10]: yes sure

File: README.md (hunk 1 above)
Comment (mheon) [reply README.md:26]: "We test that Netavark will build on the following Rust version. Newer versions should also build, and if they do not, the issue should be reported and will be fixed. Older versions are not guaranteed to build and issues will not be fixed"?

Comment (Luap99) [reply README.md:26]: sounds good

**Comment by mheon** [comment]
LGTM

**Comment by baude** [comment]
this is awesome

**Comment by edsantiago** [comment]
...and you even remembered to update the descriptions on quay, for this and the other two images. Thank you.

**Pull request merged**